src/util/tprint.cpp

```diff
@@ -52,3 +56,24 @@ i32 sntprint<>(char *buffer, u32 buf_size, const char *fmt) {
     return head;
 }
 
+#include "../test.h"
+
+TEST_FORMAT((f32) 1.5,   { .num_decimals=2 }, "1.50");
+TEST_FORMAT((f32) 1.5,   { .num_decimals=1 }, "1.5");
+TEST_FORMAT((f32) 1.45,  { .num_decimals=1 }, "1.5");
+TEST_FORMAT((f32) 1.44,  { .num_decimals=1 }, "1.4");
+
+TEST_FORMAT((f64) 1.44,  { .num_decimals=1 }, "1.4");
+TEST_FORMAT((f64) 1.44,  { .num_decimals=2 }, "1.44");
+
+TEST_FORMAT((f32) 1.5, ((FormatHint) { .num_decimals=2, .num_zero_pad=4 }), "1.50");
```

Comment:
Maybe move the cast into the macro?

Reply:
I get it to compile without the cast, which is at least a bit better.

Reply:
I don't:
```
src/util/tprint.cpp:69:27: error: expected primary-expression before ‘.’ token
   69 | TEST_FORMAT((f32) 1.5, ({ .num_decimals=2, .num_zero_pad=4 }), "1.50");
      |                           ^
src/util/../test.h:46:127: note: in definition of macro ‘TEST_CASE’
   46 | #define TEST_CASE(name, block) static int UNIQUE_NAME(_test_id_) = reg_test((name), [](GameState *game, FILE *report) -> bool block, __FILE__, __LINE__)
      |                                                                                                                               ^~~~~
src/util/tprint.cpp:69:1: note: in expansion of macro ‘TEST_FORMAT’
   69 | TEST_FORMAT((f32) 1.5, ({ .num_decimals=2, .num_zero_pad=4 }), "1.50");
```
What did you change it to?